freebites/product
# 38 update forgot password

# Description

Created an update password page which changes the user's current password and a forgot password page which sends an email with a link to the user (once the user inputs a valid email address that exists in the database), which prompts the user to change/update their password. 

We also added error checks for both the update password and the forgot password.

For the forgot password, we made sure that the email the user put in was in a valid email format and exists in the firebase. 

For the update password, we made sure that the current password matched with the "Current Password" text input, that the "New Password" was between 8 - 16 characters with at least 1 number, 1 symbol, and 1 capital, and that the "Confirm Password" matched up with the "New Password".

## Issues

Resolve #38 

## Screenshots

<img src="https://github.com/freebites/product/assets/158139019/284b139c-9e39-4b8c-9705-1bae21330eda" width="200">
<img src="https://github.com/freebites/product/assets/158139019/afa6e346-75ec-4ec1-a05b-ecce12631fe1" width="200">
<img src="https://github.com/freebites/product/assets/158139019/a5f09517-80e0-4360-ae4e-fda7fd5dff8f" width="200">
<img src="https://github.com/freebites/product/assets/158139019/7251997e-1247-4614-a816-4fc0216dd58a" width="200">
<img src="https://github.com/freebites/product/assets/158139019/da19a598-191b-4ee4-9cb5-cd04646ec368" width="200">
<img src="https://github.com/freebites/product/assets/158139019/ad75ef62-2514-42ff-95a6-30e07763012a" width="200">


## Test
Update Password:

- Log into the Account.
- Go to the Profile page by pressing the person icon on the bottom right
- Click on Settings
- Click "Update Password"
- Input the current password into the "Current Password" field
- Input your new password into the "New Password" field
- Input your new password into the "Confirmed Password" field
- If everything works (see description for errors), the password should update and the app should route you back to the home screen
- If not, the appropriate error message will appear.
- Log in to check if it has changed

Forgot Password:

- Click either "Sign In" or "Sign Up"
- Click "Forgot Password"
- Enter in a valid email that is in the firebase
- If a valid email is inputted, an "Email sent" message should appear. If not, the appropriate error message will appear.
- Check you email for the link
- Click on the link and put in another password
- Once submitted, the password should be changed
- Log in to check if it has changed

## Possible Downsides

The "New Password" and "Confirm Password" sections of the update password page update live while the "Old Password" section is only updated after the "Submit" button is pressed. We tried making it live as well but for someone reason it did not work. 

Because firebase sends the forgot password link to the user when the email is sent, the UI for the page the user is sent to when the link is pressed cannot be changed by us (it is the firebase UI). We also cannot check if the new password made by the user is valid (between 8 - 16 characters with at least 1 number, 1 symbol, and 1 capital).

## Additional Documentations
